In [1]:
import os
for f in os.listdir('../fun/'): exec(open('../fun/'+f).read())
del f

# Load data. Change winner to binary 1/0:
load( '../out/d3-fight-level-standardize-normalize.pkl' )
y[ y == -1 ] = 0
print( X.shape )

(3576, 115)


In [2]:
# train test split.
from sklearn.model_selection import train_test_split
X_train , X_test, y_train, y_test = train_test_split(
    X, y, 
    random_state = 718,
    test_size = 0.25
)

In [3]:
# Fit decision tree.
from sklearn.tree import DecisionTreeClassifier
m = DecisionTreeClassifier( max_depth = 6, min_samples_split = .1, random_state = 716 )
m.fit( X_train, y_train )

# accuracy against train data (in-model).
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( y_train, m.predict(X_train) ) )

              precision    recall  f1-score   support

           0       0.57      0.50      0.53       903
           1       0.76      0.81      0.78      1779

    accuracy                           0.70      2682
   macro avg       0.66      0.65      0.66      2682
weighted avg       0.69      0.70      0.70      2682



In [4]:
# accuracy against test data.
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( y_test, m.predict(X_test) ) )

              precision    recall  f1-score   support

           0       0.48      0.38      0.42       301
           1       0.71      0.79      0.75       593

    accuracy                           0.65       894
   macro avg       0.60      0.58      0.59       894
weighted avg       0.64      0.65      0.64       894



f1-score in-model is higher than on test indicating **over-fitting**. 

Test-set precision is still not great at 69% (predicting all-wins give 60%).